pmlc/dialect/stripe/dialect.cc

```diff
@@ -39,18 +41,19 @@ mlir::Type Dialect::parseTensor(llvm::StringRef tyData, mlir::Location loc) cons
     emitError(loc, "invalid tensor type, no ()'s on size spec");
     return Type();
   }
-  llvm::SmallVector<StringRef, 8> dims;
-  llvm::SmallVector<TensorDim, 8> odims;
+  auto dims = llvm::SmallVector<StringRef, 8>();
+  auto odims = llvm::SmallVector<TensorDim, 8>();
+  auto matches = llvm::SmallVector<StringRef, 4>();
```

Comment:
Odd style, any particular reason for incurring an extra assignment?

Reply:
Compilers are allowed to omit it (and IIRC the ones we use do); in C++17, they'll be required to.  https://en.cppreference.com/w/cpp/language/copy_elision